ur_robot_driver/doc/real_time.md

```diff
@@ -1,18 +1,44 @@
 # Setting up Ubuntu with a PREEMPT_RT kernel
 In order to run the `universal_robot_driver`, we highly recommend to setup a ubuntu system with
-real-time capabilities. Especially with a robot from the e-Series the higher control frequency
-might lead to non-smooth trajectory execution if not run using a real-time-enabled system.
+(close-to) real-time capabilities. Especially with a robot from the e-Series the higher control frequency
+might lead to connection breakdowns if not run using a real-time-enabled system.
 
 You might still be able to control the robot using a non-real-time system. This is, however, not recommended.
 
+For getting an almost real-time capable systems there are two methods available:
```

Comment:
Thank you for the feedback. Yes, writing this didn't feel too good. I deliberately chose none of the above, since I was not sure where to put the different configurations into. A PREEMPT_RT kernel-based system using SCHED_FIFO could probably called *soft real-time*, but for the lowlatency kernel I am not sure.

From my understanding *soft real-time* and *hard real-time* are more or less defined categories of real-time systems, while I wanted to state that the measures above will give you a system that comes close to the behavior of a real-time system (I guess "close to a soft real-time system") would be the better terminology here.

Reply:
Without going through the text in detail, I'd say low-latency kernel is best-effort.

`PREEMPT_RT` probably soft real-time, as you already suggested -- although there seems to be some discussion/contention around that. I'd expect companies like VxWorks and QNX to deem it *soft*, while the goal of the project has been summarised as "making Linux hard real-time capable".

Reply:
> I wanted to state that the measures above will give you a system that comes close to the behavior of a real-time system

perhaps describing the behaviour of the system would be better than trying to give it a name.

You seem to have a large set of measurements. Why not just state "with PREEMPT_RT, you get average activation latency of X, max jitter of Y, etc" (maybe you already do this)? Then link those KPIs to behaviour of the driver and/or robot in typical usage scenarios ("with PREEMPT_RT and a direct cable connection, a typical MoveIt-plus-10-other-nodes configuration should be able to achieve 500 Hz, while ..". Something like that.

Reply:
I think that's not the point I want to make there, since for those kind of analysis we have the separate document, but a description-based approach seems to be a good solution.

> PREEMPT_RT probably soft real-time, as you already suggested -- although there seems to be some discussion/contention around that. I'd expect companies like VxWorks and QNX to deem it soft, while the goal of the project has been summarised as "making Linux hard real-time capable".

Which is why I wanted to avoid such terms ;-)